File: core/src/commonMain/kotlin/work/socialhub/kmastodon/Mastodon.java
package work.socialhub.kmastodon;

import mastodon4j.api.AccountsResource;
import mastodon4j.api.AppsResource;
import mastodon4j.api.BlocksResource;
import mastodon4j.api.EmojiResource;
import mastodon4j.api.FavouritesResource;
import mastodon4j.api.FollowRequestsResource;
import mastodon4j.api.FollowsResource;
import mastodon4j.api.InstancesResource;
import mastodon4j.api.ListsResource;
import mastodon4j.api.MediaResource;
import mastodon4j.api.MutesResource;
import mastodon4j.api.NodeResource;
import mastodon4j.api.NotificationsResource;
import mastodon4j.api.OauthResource;
import mastodon4j.api.PollResource;
import mastodon4j.api.ReportsResource;
import mastodon4j.api.SearchResource;
import mastodon4j.api.StatusesResource;
import mastodon4j.api.StreamingResource;
import mastodon4j.api.TimelinesResource;
import mastodon4j.api.TrendResource;
import mastodon4j.domain.Service;

/**
 * @author hecateball
 */
public interface Mastodon extends AccountsResource, AppsResource, BlocksResource, FavouritesResource,
        FollowRequestsResource, FollowsResource, InstancesResource, MediaResource, MutesResource, NotificationsResource,
        OauthResource, ReportsResource, SearchResource, StatusesResource, StreamingResource, TimelinesResource, ListsResource,
        TrendResource, PollResource, NodeResource, EmojiResource {

    /** Get kind of service */
    public Service service();

    public AccountsResource accounts();

    public AppsResource apps();

    public BlocksResource blocks();

    public FavouritesResource favourites();

    public FollowRequestsResource followRequests();

    public FollowsResource follows();

    public InstancesResource instances();

    public MediaResource media();

    public MutesResource mutes();

    public NotificationsResource notifications();

    public OauthResource oauth();

    public ReportsResource reports();

    public SearchResource search();

    public StatusesResource statuses();

    public StreamingResource streaming();

    public TimelinesResource timelines();

    public ListsResource list();

    public TrendResource trend();

    public PollResource poll();

    public NodeResource node();

    public EmojiResource emoji();
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/Mastodon.kt
package work.socialhub.kmastodon


import work.socialhub.kmastodon.api.AccountsResource
import work.socialhub.kmastodon.api.AppsResource
import work.socialhub.kmastodon.api.BlocksResource
import work.socialhub.kmastodon.api.EmojisResource
import work.socialhub.kmastodon.api.FavouritesResource
import work.socialhub.kmastodon.api.FollowRequestsResource
import work.socialhub.kmastodon.api.FollowsResource
import work.socialhub.kmastodon.api.InstancesResource
import work.socialhub.kmastodon.api.ListsResource
import work.socialhub.kmastodon.api.MediasResource
import work.socialhub.kmastodon.api.MutesResource
import work.socialhub.kmastodon.api.NodesResource
import work.socialhub.kmastodon.api.NotificationsResource
import work.socialhub.kmastodon.api.OAuthResource
import work.socialhub.kmastodon.api.PollsResource
import work.socialhub.kmastodon.api.ReportsResource
import work.socialhub.kmastodon.api.SearchResource
import work.socialhub.kmastodon.api.StatusesResource
import work.socialhub.kmastodon.api.TimelinesResource
import work.socialhub.kmastodon.api.TrendsResource
import work.socialhub.kmastodon.domain.Service
import kotlin.js.JsExport

@JsExport
interface Mastodon {

    /** Get kind of service  */
    fun service(): Service

    fun apps(): AppsResource
    fun oauth(): OAuthResource
    fun nodes(): NodesResource
    fun emojis(): EmojisResource
    fun instances(): InstancesResource

    fun search(): SearchResource
    fun medias(): MediasResource
    fun trends(): TrendsResource
    fun lists(): ListsResource
    fun polls(): PollsResource
    fun mutes(): MutesResource
    fun blocks(): BlocksResource
    fun reports(): ReportsResource
    fun follows(): FollowsResource
    fun followRequests(): FollowRequestsResource

    fun accounts(): AccountsResource
    fun statuses(): StatusesResource
    fun timelines(): TimelinesResource
    fun favourites(): FavouritesResource
    fun notifications(): NotificationsResource

    fun uri(): String
    fun accessToken(): String
}
